src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -309,38 +330,70 @@ def _run_microtrait(genome_id, fna_file, batch_dir):
     genome_dir = os.path.join(batch_dir, genome_id)
     os.makedirs(genome_dir, exist_ok=True)
 
-    # Load the R script as an R function
-    r_script = """
-    library(microtrait)
-    extract_traits <- function(genome_file, out_dir) {
-      genome_file <- file.path(genome_file)
-      microtrait_result <- extract.traits(in_file = genome_file, out_dir = out_dir)
-      return(microtrait_result)
-    }
-    """
-    r_func = robjects.r(r_script)
-
     remove_gz_file = False
     if fna_file.endswith('.gz'):
         remove_gz_file = True
         fna_file = _unpack_gz_file(fna_file)
-    r_result = r_func(fna_file, genome_dir)
 
-    # retrieve genes_detected_table from microtrait_result and save it as a csv file
-    # genes_detected_table includes the following columns:
-    # gene_name, gene_len, hmm_name, gene_score, gene_from, gene_to, cov_gene, hmm_from, hmm_to, cov_domain
-    microtrait_result = _get_r_list_element(r_result, 'microtrait_result')
-    genes_detected_table = _get_r_list_element(microtrait_result, 'genes_detected_table')
-    data = dict()
-    for idx, name in enumerate(genes_detected_table.names):
-        data.update({name: list(genes_detected_table[idx])})
-    genes_detected_df = pd.DataFrame(data=data)
-    genes_detected_df.to_csv(os.path.join(genome_dir, 'genes_detected_table.csv'), index=False)
-
-    # remove the unpacked assembly file to save space
-    if remove_gz_file:
-        print(f'removing {fna_file}')
-        os.remove(fna_file)
+    try:
+        # Load the R script as an R function
+        r_script = """
+        library(microtrait)
+        extract_traits <- function(genome_file, out_dir) {
+          genome_file <- file.path(genome_file)
+          microtrait_result <- extract.traits(in_file = genome_file, out_dir = out_dir)
+          return(microtrait_result)
+        }
+        """
+        r_func = robjects.r(r_script)
+        r_result = r_func(fna_file, genome_dir)
+
+        microtrait_result = _get_r_list_element(r_result, 'microtrait_result')
+
+        trait_counts_atgranularity3 = _get_r_list_element(microtrait_result, 'trait_counts_atgranularity3')
+        trait_counts_atgranularity3_df = _r_table_to_df(trait_counts_atgranularity3)
+
+        trait_counts_atgranularity3_df.to_csv(os.path.join(genome_dir, 'trait_counts_atgranularity3.csv'), index=False)
```

Comment:
Can we make this so it's easy to switch granularity and the parser script Just Works regardless of the granularity? Something like

```suggestion
        trait_counts = _get_r_list_element(microtrait_result, 'trait_counts_atgranularity3')  # make this name a constant at top of file
        trait_counts = _r_table_to_df(trait_counts)

        trait_counts.to_csv(os.path.join(genome_dir, 'trait_counts.csv'), index=False)
```

Reply:
👍

Reply:
:+1: